modules/custom-status/lib/components/modals/create-edit-custom-status-modal.js

```diff
@@ -1,33 +1,64 @@
 import apiFetch from '@wordpress/api-fetch';
 import {
-	Card,
 	Button,
+	Card,
+	CardBody,
+	__experimentalDivider as Divider,
+	__experimentalHStack as HStack,
 	Modal,
 	TextControl,
 	TextareaControl,
-	Tooltip,
 	ToggleControl,
-	__experimentalDivider as Divider,
-	__experimentalHStack as HStack,
-	CardBody,
+	Tooltip,
 } from '@wordpress/components';
 import { useState } from '@wordpress/element';
 import { __, sprintf } from '@wordpress/i18n';
 
 import ErrorNotice from '../../../../shared/js/components/error-notice';
+import MetadataSelectFormTokenField from '../metadata-select-form-token-field';
 import UserSelectFormTokenField from '../user-select-form-token-field';
 
-export default function CreateEditCustomStatusModal( { customStatus, onCancel, onSuccess } ) {
+export default function CreateEditCustomStatusModal( {
+	customStatus,
+	editorialMetadatas,
+	onCancel,
+	onSuccess,
+} ) {
 	// Custom status properties
 	const [ name, setName ] = useState( customStatus?.name || '' );
 	const [ description, setDescription ] = useState( customStatus?.description || '' );
-	const [ requiredUsers, setRequiredUsers ] = useState( customStatus?.required_users || [] );
+	const [ requiredUsers, setRequiredUsers ] = useState( customStatus?.meta?.required_users || [] );
+
+	// Taxonomy conflicts arise if this is done server side, so this transient field is only set here.
+	const [ requiredMetadatas, setRequiredMetadatas ] = useState( () => {
```

Comment:
Due to the taxonomy problems, this is needed. I noticed I had some bad data sitting around so this helps to clean it up whenever an update happens. It's fine if its silent right now, but ideally we should handle this on the backend with a cron cleanup job.

Reply:
What do you mean by "taxonomy problems"/"taxonomy conflicts" here? I'm not sure what problem this is solving.

Reply:
What I originally had:

- Filter against `get_term` and `get_terms` in EM that used it's taxonomy to populate each term with the necessary `term_meta`.
- Filter against `get_term` and `get_terms` in CS that used it's taxonomy to populate each term with the necessary `term_meta`.

What ended up happening was related to init problems:

- Within EM, CS taxonomy was given back. That was rejected because we only target the EM taxonomy so no `term_meta` was populated.
- Within CS, EM taxonomy was given back. That was rejected because we only target the CS taxonomy so no `term_meta` was populated.

So the safest option to me was:

Just do this on the client side for now.

What I'd like to do ideally:

Add a GET api to CS and EM endpoints and add this functionality there. Ideally that's how all this data massaging is done and it should hopefully not have as many conflicts.